### PR TITLE
Authentification feature

### PR DESCRIPTION
**Pull request by DaryaKokorina**

### CODE REVIEW

File: apps/emloyees-management/src/app/app.module.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 @NgModule({
-  declarations: [AppComponent],
-  imports: [BrowserModule],
+  declarations: [AppComponent, HomeComponent],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    AngularFireModule.initializeApp(environment.firebase),
+    AngularFireAuthModule,
+    EmployeeAppFeatureAuthModule
```
Comment (koskosergej) [inline apps/emloyees-management/src/app/app.module.ts]: Why you import this module to  App Module? it should be available throughout the routing

File: apps/emloyees-management/src/app/home/home.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p>home component works!</p>
```
Comment (koskosergej) [inline apps/emloyees-management/src/app/home/home.component.html:1]: please remove default text

File: apps/emloyees-management/src/app/home/home.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+  selector: 'dashasorg-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.css']
+})
+export class HomeComponent {
+  constructor(private authService: AuthService, private router: Router) {}
+
+  logout() {
```
Comment (koskosergej) [inline apps/emloyees-management/src/app/home/home.component.ts:14]: will be better to create some navigation panel with the logout button

File: libs/auth/src/lib/auth.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  afterEach(() => {
+    fakeAuthState.next(null);
+  });
+
+  it('service should be created', () => {
+      expect(service).toBeTruthy();
+  });
+
+  it('should call login successfully', async () => {
```
Comment (koskosergej) [inline libs/auth/src/lib/auth.service.spec.ts]: "should call login successfully" when?

File: libs/auth/src/lib/auth.service.spec.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      await service.login(credentialsMock.email, credentialsMock.password);
+      expect(fakeAuthState.value).toEqual(resDataMock.user);
+  });
+
+  it('should fail login', async () => {
```
Comment (koskosergej) [inline libs/auth/src/lib/auth.service.spec.ts]: when should fail?

File: libs/auth/src/lib/auth.service.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+@Injectable({ providedIn: 'root' })
+export class AuthService {
+
+  constructor( // Inject Firestore service
+    private afAuth: AngularFireAuth, // Inject Firebase auth service
+    private router: Router,// NgZone service to remove outside scope warning
+  ) {}
+
+  // Sign in with email/password
+  login(email: string, password: string): Promise<string | void> {
```
Comment (koskosergej) [inline libs/auth/src/lib/auth.service.ts]: will be better to transform Promise to Observable for uniformity.

File: libs/auth/src/lib/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  // Returns true when user is looged in
+  isLoggedIn(): Observable<boolean> {
+    return this.afAuth.authState.pipe(
+      map((res) => !!res.uid)
+    );
+  }
+
+  logout(): Promise<void> {
```
Comment (koskosergej) [inline libs/auth/src/lib/auth.service.ts]: will be better to transform Promise to Observable for uniformity.

File: libs/employee-app/feature-employees-list/src/lib/employees-list-container/employees-list-container.component.html [hunk 8]
```diff
@@ -0,0 +1,2 @@
+<p>employees list container works!</p>
```
Comment (koskosergej) [inline libs/employee-app/feature-employees-list/src/lib/employees-list-container/employees-list-container.component.html:1]: please remove default text

File: libs/employee-app/ui-auth/src/lib/auth-form-container/auth-form-container.component.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  @Output() passLoginData = new EventEmitter<LoginData>();
+
+  formControlPlaceholder = 'Enter here';
+  loginData: LoginData = {
+    email: '',
+    password: ''
+  };
+
+  onSubmit() {
+    this.passLoginData.emit(this.loginData);
```
Comment (koskosergej) [inline libs/employee-app/ui-auth/src/lib/auth-form-container/auth-form-container.component.ts:19]: what happened when login data contains invalid/empty values?

**Review by koskosergej — COMMENTED** [review]
please fix comments

File: apps/emloyees-management/src/app/home/home.component.html (hunk 2 above)
Comment (DaryaKokorina) [reply apps/emloyees-management/src/app/home/home.component.html:1]: It's not so important for the current iteration. I'll do it through the next steps.

File: libs/auth/src/lib/auth.service.spec.ts (hunk 4 above)
Comment (DaryaKokorina) [reply libs/auth/src/lib/auth.service.spec.ts]: fixed

File: libs/auth/src/lib/auth.service.spec.ts (hunk 5 above)
Comment (DaryaKokorina) [reply libs/auth/src/lib/auth.service.spec.ts]: fixed

File: libs/auth/src/lib/auth.service.ts (hunk 6 above)
Comment (DaryaKokorina) [reply libs/auth/src/lib/auth.service.ts]: fixed

File: libs/auth/src/lib/auth.service.ts (hunk 7 above)
Comment (DaryaKokorina) [reply libs/auth/src/lib/auth.service.ts]: fixed

File: apps/emloyees-management/src/app/app.module.ts (hunk 1 above)
Comment (DaryaKokorina) [reply apps/emloyees-management/src/app/app.module.ts]: fixed

File: apps/emloyees-management/src/app/home/home.component.ts (hunk 3 above)
Comment (DaryaKokorina) [reply apps/emloyees-management/src/app/home/home.component.ts:14]: next iteration

File: libs/employee-app/feature-employees-list/src/lib/employees-list-container/employees-list-container.component.html (hunk 8 above)
Comment (DaryaKokorina) [reply libs/employee-app/feature-employees-list/src/lib/employees-list-container/employees-list-container.component.html:1]: next iteration

File: libs/employee-app/ui-auth/src/lib/auth-form-container/auth-form-container.component.ts (hunk 9 above)
Comment (DaryaKokorina) [reply libs/employee-app/ui-auth/src/lib/auth-form-container/auth-form-container.component.ts:19]: added login form validation and error handling

**Pull request merged**